src/utility/CurationUtility.js

```diff
@@ -253,3 +274,218 @@ export const prepareInputForDataSubmission = (input) => {
+  let variantTypes = deprepare_input.variant_types;
+  let pub_data = Object.keys(publicationsObj);
+  for (const item of VariantTypesAttribs) {
+    variantTypesObj[item.primaryType.inputKey] = {};
+    for (const secondaryTypeObj of item.secondaryType) {
+      variantTypesObj[item.primaryType.inputKey][secondaryTypeObj.inputKey] = {
+        nmd_escape: false,
+        de_novo: false,
+        inherited: false,
+        unknown_inheritance: false,
+        supporting_papers: pub_data,
+        comment: "",
+      };
+    }
+  }
+
+  variantTypes.forEach((varianttypes) => {
+    const {
+      comment,
+      de_novo,
+      inherited,
+      nmd_escape,
+      primary_type,
+      secondary_type,
+      supporting_papers,
+      unknown_inheritance,
+    } = varianttypes;
+
+    // Ensure the primary_type key exists
+    if (variantTypesObj[primary_type][secondary_type]) {
+      variantTypesObj[primary_type][secondary_type] = {
+        comment,
+        de_novo,
+        inherited,
+        nmd_escape,
+        supporting_papers,
+        unknown_inheritance,
+      };
+    }
+  });
+
+  let MechanismNameObj = {
+    name: "",
+    support: "",
+  };
+  let mechanism_name = deprepare_input.molecular_mechanism;
+
+  if (mechanism_name.name) {
+    (MechanismNameObj.name = mechanism_name.name),
+      (MechanismNameObj.support = mechanism_name.support);
+  }
+
+  let MechanismSynopsisObj = {
+    name: "",
+    support: "",
+  };
+  let mechanismSynopsis = deprepare_input.mechanism_synopsis;
+
+  if (mechanismSynopsis.name) {
+    (MechanismSynopsisObj.name = mechanismSynopsis.name),
+      (MechanismSynopsisObj.support = mechanismSynopsis.support);
+  }
+
+  let MechanismEvidenceObj = {};
+  let MechanismEvidence = deprepare_input.mechanism_evidence;
+  if (MechanismEvidence) {
+  }
+
+  for (const key of Object.keys(publicationsObj)) {
+    MechanismEvidenceObj[key] = {
+      description: "",
+      evidence_types: {},
+    };
+  }
+
+  // Populate the evidence_types for each key
+  for (const key of Object.keys(MechanismEvidenceObj)) {
+    for (const item of EvidenceTypesAttribs) {
+      if (!MechanismEvidenceObj[key].evidence_types[item.primaryType]) {
+        MechanismEvidenceObj[key].evidence_types[item.primaryType] = [];
+      }
+      MechanismEvidenceObj[key].evidence_types[item.primaryType].push(
+        item.secondaryType
+      );
+    }
+  }
+
+  if (MechanismEvidence) {
+    MechanismEvidence.forEach((evidence) => {
+      const pmid = evidence.pmid;
+
+      if (pmid) {
+        MechanismEvidenceObj[pmid] = {
+          description: evidence.description,
+          evidence_types: {},
+        };
+
+        if (!MechanismEvidenceObj[pmid].evidence_types[evidence.primaryType]) {
+          MechanismEvidenceObj[pmid].evidence_types[evidence.primaryType] = [];
+        }
+
+        MechanismEvidenceObj[pmid].evidence_types[evidence.primaryType].push(
+          evidence.secondaryType
+        );
+      }
+    });
+  }
```

Comment:
the evidence object will have evidence_types array which has to be iterated but it seems like we are not iterating over the array, please have a look at this logic